src/react/connect.ts

```diff
@@ -0,0 +1,29 @@
+import { useStream, NOT_YET_EMITTED } from './useStream';
+import { Observable } from 'rxjs';
+import React, { ComponentType } from 'react';
+
+/**
+ * Higher order component for connecting a component to a stream
+ *
+ * The stream does not need to provide all the properties the component expects,
+ * any missing properties will be required by the component returned from this
+ * component.
+ *
+ * @param WrappedComponent Component that will recieve props from the stream
+ * @param stream$ Stream that will provide props to the component
+ * @see useStream
+ */
+export const connect = <Props, Observed extends Partial<Props>>(
+  Component: ComponentType<Props>,
+  stream$: Observable<Observed>
+) => (props: Omit<Props, keyof Observed>) => {
+  const value = useStream(stream$);
+
+  if (value === NOT_YET_EMITTED) return null;
+
+  // Typescript doesn't recognize this as Observed & Props for some reason
+  // Question on StackOverflow: https://stackoverflow.com/q/60758084/1104307
+  const newProps = { ...props, ...value } as Props & Observed;
```

Comment:
Isn't the precedence in the current `connect` (from `streamUtils`) the opposite? (`...value, ...props`)

Reply:
Nice catch @holographic-principle.

Personally, I don't see a strong case for why "props" or "streamedProps" should take precedence 🤷‍♂ Did you switch these around for any particular reason @tlaundal ? 

In general, I can't think of a case where having the same key defined in "props" and "streamedProps" is a good pattern as I think this could cause a lot of confusion. Perhaps we could add a warning in dev mode?

Reply:
So actually, this kind of shadowing is actually caught by TS because of the types (this is what the `Omit` does), so I don't really think this is a problem. 

I can change the order to be the same as in ardoq-front, just to be on the safe side, but I don't think we should document or test this, as it is explicitly forbidden by the types.

Reply:
Ok, good 👍